### ISSUE TITLE
Shared memory allocation example (Tutorials/Gettingstarted/fpga_compile/part2_dpcpp_functor_usm)  not working  on Intel N6001 board  

### ISSUE THREAD

**Issue by umairsiddiqui-digitek** [OFS/oneapi-asp#140]
i am able to run board_test application
https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C%2B%2BSYCL_FPGA/ReferenceDesigns/board_test

which has USM memory test. 

But I am not able to run the Shared memory allocation example on Intel N6001 board 

- Tutorials/Gettingstarted/fpga_compile/part2_dpcpp_functor_usm

https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/GettingStarted/fpga_compile/part2_dpcpp_functor_usm

- Tutorials/GettingStarted/fpga_compile/part3_dpcpp_lambda_usm

https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/GettingStarted/fpga_compile/part3_dpcpp_lambda_usm

the code allocate shared buffers
```
part2_dpcpp_functor_usm/src/vector_add.cpp:58:    // allocate in shared memory so the kernel can see them
part2_dpcpp_functor_usm/src/vector_add.cpp:59:    int *vec_a = sycl::malloc_shared<int>(kVectSize, q);
part2_dpcpp_functor_usm/src/vector_add.cpp:60:    int *vec_b = sycl::malloc_shared<int>(kVectSize, q);
part2_dpcpp_functor_usm/src/vector_add.cpp:61:    int *vec_c = sycl::malloc_shared<int>(kVectSize, q);
```

code seems to get stuck when executing

https://github.com/oneapi-src/oneAPI-samples/blob/d9ea61ab122ab030fbd74362bf9cd68766556117/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/GettingStarted/fpga_compile/part2_dpcpp_functor_usm/src/vector_add.cpp#L69

and 

https://github.com/oneapi-src/oneAPI-samples/blob/d9ea61ab122ab030fbd74362bf9cd68766556117/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/GettingStarted/fpga_compile/part3_dpcpp_lambda_usm/src/vector_add.cpp#L63

 

i am using OFS 2024.1-1 and repo tags mentioned here
https://github.com/orgs/OFS/discussions/60

https://github.com/oneapi-src/oneAPI-samples (tag 2024.1.0)

$ quartus_sh --version
Quartus Prime Shell
Version 23.4.0 Build 79 11/22/2023 Patches 0.02iofs,0.17 SC Pro Edition
Copyright (C) 2023  Intel Corporation. All rights reserved.

$ icpx --version
Intel(R) oneAPI DPC++/C++ Compiler 2024.0.2 (2024.0.2.20231213)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/intel/oneapi/compiler/2024.0/bin/compiler
Configuration file: /opt/intel/oneapi/compiler/2024.0/bin/compiler/../icpx.cfg

**Comment by aktole** [comment]
Hi, 

[oneAPI code samples repository](https://github.com/oneapi-src/oneAPI-samples) tag [must match the oneAPI _Compiler_ version](https://ofs.github.io/ofs-2024.1-1/hw/common/user_guides/oneapi_asp/ug_oneapi_asp/#26-compile-and-run-oneapi-sample-applications) you are using (2024.0 from your icpx --version output). 
Can you try with the correct code sample repository tag (i.e. 2024.0 for oneAPI compiler version 2024.0.x)? 
If you would like to use the latest oneAPI compiler and code samples tag (i.e. 2024.1), please download the latest Intel® oneAPI Base Toolkit version 2024.1 from [here](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html). 
Let us know if you continue seeing this issue with the matching compiler and code sample repository tag.

Thanks

**Comment by umairsiddiqui-digitek** [comment]
i resynthesize the tutorial with 
https://github.com/oneapi-src/oneAPI-samples/releases/tag/2024.0.0

but i am getting same results. 

i also filed https://github.com/OFS/oneapi-asp/issues/141

but this is a small example so memory shouldn't be problem.

**Comment by mfmendoz-Intel** [comment]
Hi,
sorry for the delay with response, we were able to reproduce this issue, we are looking at this internally and will get back to you.

**Comment by mfmendoz-Intel** [comment]
Hi @umairsiddiqui-digitek,

The issue has been resolved in the lastest oneAPI version [2024.2.1](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html?operatingsystem=linux&linux-install-type=offline), please do the upgrade to this version and use the corresponding tag [2024.2.0](https://github.com/oneapi-src/oneAPI-samples/tree/2024.2.0) for the _Tutorials/Gettingstarted/fpga_compile/_ example.

Thanks
